compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -1205,6 +1205,14 @@ fn compare_number_of_generics<'tcx>(
         return Ok(());
     }
 
+    // Skip emitting errors for RPITITs, we've already checked generics for trait's and impl's
+    // functions.
```

Comment:
```suggestion
    // We never need to emit a separate error for RPITITs, since if an RPITIT
    // has mismatched type or const generic arguments, then the method that it's
    // inheriting the generics from will also have mismatched arguments, and
    // we'll report an error for that instead. Delay a bug for safety, though.
```